packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -262,19 +297,19 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
    * @internal
    */
   protected _renderTask(): any {
+
+    const isFeatureFlagEnabled = FeatureFlags.of(this).isEnabled(cxapi.USE_NEW_S3URI_PARAMETERS_FOR_BEDROCK_INVOKE_MODEL_TASK);
+    const inputSource = this.getInputSource(this.props.input, this.props.inputPath, isFeatureFlagEnabled);
+    const outputSource = this.getOutputSource(this.props.output, this.props.outputPath, isFeatureFlagEnabled);
     return {
       Resource: integrationResourceArn('bedrock', 'invokeModel'),
       Parameters: sfn.FieldUtils.renderObject({
         ModelId: this.props.model.modelArn,
         Accept: this.props.accept,
         ContentType: this.props.contentType,
         Body: this.props.body?.value,
-        Input: this.props.input?.s3Location ? {
-          S3Uri: `s3://${this.props.input.s3Location.bucketName}/${this.props.input.s3Location.objectKey}`,
-        } : this.props.inputPath ? { S3Uri: this.props.inputPath } : undefined,
-        Output: this.props.output?.s3Location ? {
-          S3Uri: `s3://${this.props.output.s3Location.bucketName}/${this.props.output.s3Location.objectKey}`,
-        } : this.props.outputPath ? { S3Uri: this.props.outputPath }: undefined,
+        Input: this.props.input || (this.props.inputPath && !isFeatureFlagEnabled) ? { S3Uri: inputSource } : undefined,
+        Output: this.props.output || ( this.props.outputPath && !isFeatureFlagEnabled) ? { S3Uri: outputSource } : undefined,
```

Comment:
should you use the `inputSource` and `outputSource` you defined in lines 302, and 303